crates/ruff/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -57,26 +57,28 @@ pub(crate) fn manual_list_comprehension(checker: &mut Checker, target: &Expr, bo
         return;
     };
 
-    let (stmt, conditional) = match body {
+    let (stmt, conditional, if_test) = match body {
         // ```python
         // for x in y:
         //     if z:
         //         filtered.append(x)
         // ```
-        [Stmt::If(ast::StmtIf { body, orelse, .. })] => {
+        [Stmt::If(ast::StmtIf {
+            body, orelse, test, ..
+        })] => {
             if !orelse.is_empty() {
                 return;
             }
             let [stmt] = body.as_slice() else {
                 return;
             };
-            (stmt, true)
+            (stmt, true, Some(test))
         }
         // ```python
         // for x in y:
         //     filtered.append(f(x))
         // ```
-        [stmt] => (stmt, false),
+        [stmt] => (stmt, false, None),
```

Comment:
Can we get rid of the true/false that’s part of this tuple, since we can now test for some / none?